atc/db/resource_cache.go

```diff
@@ -241,6 +242,26 @@ func (cache *usedResourceCache) BaseResourceType() *UsedBaseResourceType {
 	return cache.resourceConfig.CreatedByResourceCache().BaseResourceType()
 }
 
+func (cache *usedResourceCache) LoadVersionMetadata() ([]atc.MetadataField, error) {
+	r := &resourceConfigVersion{conn: cache.conn}
+
+	versionJSON, _ := json.Marshal(cache.Version())
+	row := resourceConfigVersionQuery.
+		Where(sq.Expr("v.version_md5 = md5(?)", versionJSON)).
```

Comment:
You would also need to filter on `resource_config_scope_id` because there can be multiple versions that have the same version strings but belong to different resource config scopes (which can result in different metadata)

Reply:
Good point. Will add it.

Reply:
@clarafu Please take a look again.